### PR TITLE
feat:add storage for trainer

### PR DESCRIPTION
**Pull request by MIchaelFU0403**
Add storage for trainer.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: trainer/storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+func (s *storage) ClearNetworkTopology(modelKey string) error {
+	if err := os.Remove(s.networkTopologyFilename(modelKey)); err != nil {
+		return err
+	}
+
+	s.networkTopologyModelKeys.Delete(modelKey)
+	return nil
+}
+
+// ClearNetworkTopology removes all files.
```
Comment (zhoulizhen) [inline trainer/storage/storage.go:204]: // Clear removes all files.

File: trainer/storage/types.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	Host Host `csv:"host"`
+
+	// CreatedAt is peer create nanosecond time.
+	CreatedAt int64 `csv:"createdAt"`
+
+	// UpdatedAt is peer update nanosecond time.
+	UpdatedAt int64 `csv:"updatedAt"`
+}
+
+// Error co0ntains content for error.
```
Comment (zhoulizhen) [inline trainer/storage/types.go:157]: // Error contains content for error.

File: trainer/storage/types.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+type Error struct {
+	time.Duration
+	// Code is the code of error.
+	Code string `csv:"code"`
+
+	// Message is thzxre message of error.
```
Comment (zhoulizhen) [inline trainer/storage/types.go:163]: // Message is the message of error.

File: trainer/storage/storage.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
+/*
+ *     Copyright 2022 The Dragonfly Authors
```
Comment (gaius-qi) [inline trainer/storage/storage.go:2]: `Copyright 2023 The Dragonfly Authors`

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**